### ISSUE TITLE
[CORTEX]::Error: ggml_cuda_host_malloc: failed to allocate 1034.00 MiB of pinned memory: no CUDA-capable device is detected

### ISSUE THREAD

**Issue by avinashtrivedi** [janhq/jan#3782]
### Jan version

1

### Describe the Bug

My System (pop os) is having 32GB RAM, 12 GB RTX 4080. for any model even for  a singe layer offloading on GPU, i am getting below error although BLAS = 1 and using local host llama.cpp i am able to utilize GPU.

2024-10-11T12:45:53.660Z [CORTEX]::Error: ggml_cuda_init: failed to initialize CUDA: no CUDA-capable device is detected
llm_load_tensors: ggml ctx size =    0.14 MiB

2024-10-11T12:45:54.042Z [CORTEX]::Error: llm_load_tensors: offloading 13 repeating layers to GPU
llm_load_tensors: offloaded 13/33 layers to GPU
llm_load_tensors:        CPU buffer size =  8137.64 MiB
........................................................................
2024-10-11T12:45:54.043Z [CORTEX]::Error: .................

2024-10-11T12:45:54.053Z [CORTEX]::Error: llama_new_context_with_model: n_ctx      = 4096
llama_new_context_with_model: n_batch    = 2048
llama_new_context_with_model: n_ubatch   = 2048
llama_new_context_with_model: flash_attn = 1
llama_new_context_with_model: freq_base  = 500000.0
llama_new_context_with_model: freq_scale = 1
ggml_cuda_host_malloc: failed to allocate 512.00 MiB of pinned memory: no CUDA-capable device is detected

2024-10-11T12:45:54.309Z [CORTEX]::Error: llama_kv_cache_init:        CPU KV buffer size =   512.00 MiB
llama_new_context_with_model: KV self size  =  512.00 MiB, K (f16):  256.00 MiB, V (f16):  256.00 MiB
ggml_cuda_host_malloc: failed to allocate 0.49 MiB of pinned memory: no CUDA-capable device is detected
llama_new_context_with_model:        CPU  output buffer size =     0.49 MiB

2024-10-11T12:45:54.310Z [CORTEX]::Error: ggml_cuda_host_malloc: failed to allocate 1034.00 MiB of pinned memory: no CUDA-capable device is detected
llama_new_context_with_model:  CUDA_Host compute buffer size =  1034.00 MiB
llama_new_context_with_model: graph nodes  = 903
llama_new_context_with_model: graph splits = 1

### Screenshots / Logs

{"timestamp":1728651430,"level":"INFO","function":"LoadModelImpl","line":413,"message":"system info","n_threads":24,"total_threads":32,"system_info":"AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | "}

2024-10-11T12:57:10.160Z [CORTEX]::Error: llama_model_loader: loaded meta data with 33 key-value pairs and 292 tensors from /home/sarang/Downloads/Meta-Llama-3.1-8B-Instruct-Q8_0.gguf (version GGUF V3 (latest))
llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
llama_model_loader: - kv   0:                       general.architecture str              = llama
llama_model_loader: - kv   1:                               general.type str              = model
llama_model_loader: - kv   2:                               general.name str              = Meta Llama 3.1 8B Instruct
llama_model_loader: - kv   3:                           general.finetune str              = Instruct
llama_model_loader: - kv   4:                           general.basename str              = Meta-Llama-3.1
llama_model_loader: - kv   5:                         general.size_label str              = 8B
llama_model_loader: - kv   6:                            general.license str              = llama3.1
llama_model_loader: - kv   7:                               general.tags arr[str,6]       = ["facebook", "meta", "pytorch", "llam...
llama_model_loader: - kv   8:                          general.languages arr[str,8]       = ["en", "de", "fr", "it", "pt", "hi", ...
llama_model_loader: - kv   9:                          llama.block_count u32              = 32
llama_model_loader: - kv  10:                       llama.context_length u32              = 131072
llama_model_loader: - kv  11:                     llama.embedding_length u32              = 4096
llama_model_loader: - kv  12:                  llama.feed_forward_length u32              = 14336
llama_model_loader: - kv  13:                 llama.attention.head_count u32              = 32
llama_model_loader: - kv  14:              llama.attention.head_count_kv u32              = 8
llama_model_loader: - kv  15:                       llama.rope.freq_base f32              = 500000.000000
llama_model_loader: - kv  16:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010
llama_model_loader: - kv  17:                          general.file_type u32              = 7
llama_model_loader: - kv  18:                           llama.vocab_size u32              = 128256
llama_model_loader: - kv  19:                 llama.rope.dimension_count u32              = 128
llama_model_loader: - kv  20:                       tokenizer.ggml.model str              = gpt2
llama_model_loader: - kv  21:                         tokenizer.ggml.pre str              = llama-bpe

2024-10-11T12:57:10.176Z [CORTEX]::Error: llama_model_loader: - kv  22:                      tokenizer.ggml.tokens arr[str,128256]  = ["!", "\"", "#", "$", "%", "&", "'", ...

2024-10-11T12:57:10.182Z [CORTEX]::Error: llama_model_loader: - kv  23:                  tokenizer.ggml.token_type arr[i32,128256]  = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...

2024-10-11T12:57:10.215Z [CORTEX]::Error: llama_model_loader: - kv  24:                      tokenizer.ggml.merges arr[str,280147]  = ["Ġ Ġ", "Ġ ĠĠĠ", "ĠĠ ĠĠ", "...
llama_model_loader: - kv  25:                tokenizer.ggml.bos_token_id u32              = 128000
llama_model_loader: - kv  26:                tokenizer.ggml.eos_token_id u32              = 128009
llama_model_loader: - kv  27:                    tokenizer.chat_template str              = {{- bos_token }}\n{%- if custom_tools ...
llama_model_loader: - kv  28:               general.quantization_version u32              = 2
llama_model_loader: - kv  29:                      quantize.imatrix.file str              = /models_out/Meta-Llama-3.1-8B-Instruc...
llama_model_loader: - kv  30:                   quantize.imatrix.dataset str              = /training_dir/calibration_datav3.txt
llama_model_loader: - kv  31:             quantize.imatrix.entries_count i32              = 224
llama_model_loader: - kv  32:              quantize.imatrix.chunks_count i32              = 125
llama_model_loader: - type  f32:   66 tensors
llama_model_loader: - type q8_0:  226 tensors

2024-10-11T12:57:10.529Z [CORTEX]::Error: llm_load_vocab: special tokens cache size = 256

2024-10-11T12:57:10.584Z [CORTEX]::Error: llm_load_vocab: token to piece cache size = 0.7999 MB
llm_load_print_meta: format           = GGUF V3 (latest)
llm_load_print_meta: arch             = llama
llm_load_print_meta: vocab type       = BPE
llm_load_print_meta: n_vocab          = 128256
llm_load_print_meta: n_merges         = 280147
llm_load_print_meta: vocab_only       = 0
llm_load_print_meta: n_ctx_train      = 131072
llm_load_print_meta: n_embd           = 4096
llm_load_print_meta: n_layer          = 32
llm_load_print_meta: n_head           = 32
llm_load_print_meta: n_head_kv        = 8
llm_load_print_meta: n_rot            = 128
llm_load_print_meta: n_swa            = 0
llm_load_print_meta: n_embd_head_k    = 128
llm_load_print_meta: n_embd_head_v    = 128
llm_load_print_meta: n_gqa            = 4
llm_load_print_meta: n_embd_k_gqa     = 1024
llm_load_print_meta: n_embd_v_gqa     = 1024
llm_load_print_meta: f_norm_eps       = 0.0e+00
llm_load_print_meta: f_norm_rms_eps   = 1.0e-05
llm_load_print_meta: f_clamp_kqv      = 0.0e+00
llm_load_print_meta: f_max_alibi_bias = 0.0e+00
llm_load_print_meta: f_logit_scale    = 0.0e+00
llm_load_print_meta: n_ff             = 14336
llm_load_print_meta: n_expert         = 0
llm_load_print_meta: n_expert_used    = 0
llm_load_print_meta: causal attn      = 1
llm_load_print_meta: pooling type     = 0
llm_load_print_meta: rope type        = 0
llm_load_print_meta: rope scaling     = linear
llm_load_print_meta: freq_base_train  = 500000.0
llm_load_print_meta: freq_scale_train = 1
llm_load_print_meta: n_ctx_orig_yarn  = 131072
llm_load_print_meta: rope_finetuned   = unknown
llm_load_print_meta: ssm_d_conv       = 0
llm_load_print_meta: ssm_d_inner      = 0
llm_load_print_meta: ssm_d_state      = 0
llm_load_print_meta: ssm_dt_rank      = 0
llm_load_print_meta: model type       = 8B
llm_load_print_meta: model ftype      = Q8_0
llm_load_print_meta: model params     = 8.03 B
llm_load_print_meta: model size       = 7.95 GiB (8.50 BPW) 
llm_load_print_meta: general.name     = Meta Llama 3.1 8B Instruct
llm_load_print_meta: BOS token        = 128000 '<|begin_of_text|>'
llm_load_print_meta: EOS token        = 128009 '<|eot_id|>'
llm_load_print_meta: LF token         = 128 'Ä'
llm_load_print_meta: EOT token        = 128009 '<|eot_id|>'
llm_load_print_meta: max token length = 256

2024-10-11T12:57:10.606Z [CORTEX]::Error: ggml_cuda_init: failed to initialize CUDA: no CUDA-capable device is detected
llm_load_tensors: ggml ctx size =    0.14 MiB

2024-10-11T12:57:10.982Z [CORTEX]::Error: llm_load_tensors: offloading 3 repeating layers to GPU
llm_load_tensors: offloaded 3/33 layers to GPU
llm_load_tensors:        CPU buffer size =  8137.64 MiB
................................................................
2024-10-11T12:57:10.983Z [CORTEX]::Error: .........................

2024-10-11T12:57:10.994Z [CORTEX]::Error: llama_new_context_with_model: n_ctx      = 4096
llama_new_context_with_model: n_batch    = 2048
llama_new_context_with_model: n_ubatch   = 2048
llama_new_context_with_model: flash_attn = 1
llama_new_context_with_model: freq_base  = 500000.0
llama_new_context_with_model: freq_scale = 1
ggml_cuda_host_malloc: failed to allocate 512.00 MiB of pinned memory: no CUDA-capable device is detected

2024-10-11T12:57:11.250Z [CORTEX]::Error: llama_kv_cache_init:        CPU KV buffer size =   512.00 MiB
llama_new_context_with_model: KV self size  =  512.00 MiB, K (f16):  256.00 MiB, V (f16):  256.00 MiB
ggml_cuda_host_malloc: failed to allocate 0.49 MiB of pinned memory: no CUDA-capable device is detected
llama_new_context_with_model:        CPU  output buffer size =     0.49 MiB

2024-10-11T12:57:11.251Z [CORTEX]::Error: ggml_cuda_host_malloc: failed to allocate 1034.00 MiB of pinned memory: no CUDA-capable device is detected
llama_new_context_with_model:  CUDA_Host compute buffer size =  1034.00 MiB
llama_new_context_with_model: graph nodes  = 903
llama_new_context_with_model: graph splits = 1

2024-10-11T12:57:13.307Z [CORTEX]:: Load model success with response {}
2024-10-11T12:57:13.307Z [CORTEX]:: Validating model Meta-Llama-3.1-8B-Instruct-Q8_0
2024-10-11T12:57:13.310Z [CORTEX]:: Validate model state with response 200
2024-10-11T12:57:13.312Z [CORTEX]:: Validate model state success with response {"model_data":"{\"frequency_penalty\":0.0,\"grammar\":\"\",\"ignore_eos\":false,\"logit_bias\":[],\"min_p\":0.05000000074505806,\"mirostat\":0,\"mirostat_eta\":0.10000000149011612,\"mirostat_tau\":5.0,\"model\":\"/home/sarang/Downloads/Meta-Llama-3.1-8B-Instruct-Q8_0.gguf\",\"n_ctx\":4096,\"n_keep\":0,\"n_predict\":2,\"n_probs\":0,\"penalize_nl\":false,\"penalty_prompt_tokens\":[],\"presence_penalty\":0.0,\"repeat_last_n\":64,\"repeat_penalty\":1.0,\"seed\":4294967295,\"stop\":[],\"stream\":false,\"temperature\":0.800000011920929,\"tfs_z\":1.0,\"top_k\":40,\"top_p\":0.949999988079071,\"typical_p\":1.0,\"use_penalty_prompt_tokens\":false}","model_loaded":true}
2024-10-11T12:57:31.890Z [CORTEX]:: 20241011 12:57:11.480607 UTC 33067 DEBUG [Initialize] Available slots:  - llama_server_context.cc:225
20241011 12:57:11.480620 UTC 33067 DEBUG [Initialize]  -> Slot 0 - max context: 4096 - llama_server_context.cc:233
20241011 12:57:11.480797 UTC 33067 INFO  Started background task here! - llama_server_context.cc:252
20241011 12:57:11.480913 UTC 33067 INFO  Warm-up model: Meta-Llama-3.1-8B-Instruct-Q8_0 - llama_engine.cc:819
20241011 12:57:11.481128 UTC 33118 DEBUG [LaunchSlotWithData] slot 0 is processing [task id: 0] - llama_server_context.cc:623
20241011 12:57:11.481196 UTC 33118 INFO  kv cache rm [p0, end) -  id_slot: 0, task_id: 0, p0: 0 - llama_server_context.cc:1544
20241011 12:57:13.299702 UTC 33118 DEBUG [PrintTimings] PrintTimings: prompt eval time = 753.196ms / 2 tokens (376.598 ms per token, 2.65535132953 tokens per second) - llama_client_slot.cc:79
20241011 12:57:13.299720 UTC 33118 DEBUG [PrintTimings] PrintTimings:        eval time = 1065.36 ms / 4 runs   (266.34 ms per token, 3.75459938425 tokens per second)
 - llama_client_slot.cc:86
20241011 12:57:13.299723 UTC 33118 DEBUG [PrintTimings] PrintTimings:       total time = 1818.556 ms - llama_client_slot.cc:92
20241011 12:57:13.299816 UTC 33118 INFO  slot released: id_slot: 0, id_task: 0, n_ctx: 4096, n_past: 6, n_system_tokens: 0, n_cache_tokens: 0, truncated: 0 - llama_server_context.cc:1304
20241011 12:57:13.299818 UTC 33118 DEBUG [UpdateSlots] all slots are idle and system prompt is empty, clear the KV cache - llama_server_context.cc:1250
20241011 12:57:13.299819 UTC 33118 DEBUG [KvCacheClear] Clear the entire KV cache - llama_server_context.cc:258
20241011 12:57:13.299955 UTC 33067 INFO  {"content":", my dear readers!","generation_settings":{"frequency_penalty":0.0,"grammar":"","ignore_eos":false,"logit_bias":[],"min_p":0.05000000074505806,"mirostat":0,"mirostat_eta":0.10000000149011612,"mirostat_tau":5.0,"model":"/home/sarang/Downloads/Meta-Llama-3.1-8B-Instruct-Q8_0.gguf","n_ctx":4096,"n_keep":0,"n_predict":2,"n_probs":0,"penalize_nl":false,"penalty_prompt_tokens":[],"presence_penalty":0.0,"repeat_last_n":64,"repeat_penalty":1.0,"seed":4294967295,"stop":[],"stream":false,"temperature":0.800000011920929,"tfs_z":1.0,"top_k":40,"top_p":0.949999988079071,"typical_p":1.0,"use_penalty_prompt_tokens":false},"model":"/home/sarang/Downloads/Meta-Llama-3.1-8B-Instruct-Q8_0.gguf","prompt":"Hello","slot_id":0,"stop":true,"stopped_eos":false,"stopped_limit":true,"stopped_word":false,"stopping_word":"","timings":{"predicted_ms":1065.36,"predicted_n":4,"predicted_per_second":3.7545993842457013,"predicted_per_token_ms":266.34,"prompt_ms":753.196,"prompt_n":2,"prompt_per_second":2.6553513295344104,"prompt_per_token_ms":376.598},"tokens_cached":6,"tokens_evaluated":2,"tokens_predicted":4,"truncated":false} - llama_engine.cc:827
20241011 12:57:13.300063 UTC 33067 INFO  Model loaded successfully: Meta-Llama-3.1-8B-Instruct-Q8_0 - llama_engine.cc:216
20241011 12:57:13.309911 UTC 33069 INFO  Model status responded - llama_engine.cc:259
20241011 12:57:13.317150 UTC 33064 INFO  Request 1, model Meta-Llama-3.1-8B-Instruct-Q8_0: Generating response for inference request - llama_engine.cc:469
20241011 12:57:13.317159 UTC 33064 INFO  Request 1: Stop words:[
	"<|eot_id|>"
]
 - llama_engine.cc:486
20241011 12:57:13.317198 UTC 33064 INFO  Request 1: Streamed, waiting for respone - llama_engine.cc:608
20241011 12:57:13.317245 UTC 33064 DEBUG [makeHeaderString] send stream with transfer-encoding chunked - HttpResponseImpl.cc:535
20241011 12:57:13.342451 UTC 33118 DEBUG [LaunchSlotWithData] slot 0 is processing [task id: 1] - llama_server_context.cc:623
20241011 12:57:13.343034 UTC 33118 INFO  kv cache rm [p0, end) -  id_slot: 0, task_id: 1, p0: 0 - llama_server_context.cc:1544
20241011 12:57:31.890354 UTC 33118 DEBUG [PrintTimings] PrintTimings: prompt eval time = 9360.546ms / 399 tokens (23.4600150376 ms per token, 42.625718628 tokens per second) - llama_client_slot.cc:79
20241011 12:57:31.890370 UTC 33118 DEBUG [PrintTimings] PrintTimings:        eval time = 9187.343 ms / 55 runs 

### What is your OS?

- [ ] MacOS
- [ ] Windows
- [X] Linux
